### PR TITLE
Parsing

### PR DESCRIPTION
**Pull request by tavisca-ashwini**
@ytokas : Kindly check code for Parsing

### CODE REVIEW

**Comment by tavisca-ashwini** [comment]
@ytokas : Kindly review the code for Parsing.

**Comment by tavisca-ashwini** [comment]
@: Kindly review my code.

**Comment by tavisca-ashwini** [comment]
@ytokas  : Kindly review my code.

File: OperatorOverloading/01 Operator Overloading/ClassLibrary1/FileExchangeRateProvider.cs [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+using System.Text;
+using System.Threading.Tasks;
+using OperatorOverloading.Parse;
+using System.Configuration;
+
+namespace OperatorOverloading.Dbl
+{
+    public class FileExchangeRateProvider:IExchangeRateProvider
+    {
+         static string[] initialseperatedString;
```
Comment (ytokas) [inline OperatorOverloading/01 Operator Overloading/ClassLibrary1/FileExchangeRateProvider.cs:13]: Why is it static?

File: OperatorOverloading/01 Operator Overloading/ClassLibrary1/FileExchangeRateProvider.cs [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+         static string[] currencySeperatedString;
+
+        public  double GetExchangeRate(string sourceCurrency,string targetCurrency)
+        {
+            string line;
+            string completeData = "";
+
+            System.IO.StreamReader file = new System.IO.StreamReader(ConfigurationManager.AppSettings["Path"]);
+            while ((line = file.ReadLine()) != null)
```
Comment (ytokas) [inline OperatorOverloading/01 Operator Overloading/ClassLibrary1/FileExchangeRateProvider.cs:22]: check out File.ReadAllText

File: OperatorOverloading/01 Operator Overloading/ClassLibrary1/FileExchangeRateProvider.cs [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+
+            currencySeperatedString = initialseperatedString[2].Split(',');
+
+            double converter1 = GetConverter(sourceCurrency);
+            double converter2 = GetConverter(targetCurrency);
+
+            return (converter2/converter1);
+        }
+        public static double GetConverter(string sourceCurrency)
+         {
```
Comment (ytokas) [inline OperatorOverloading/01 Operator Overloading/ClassLibrary1/FileExchangeRateProvider.cs:39]: Read about exchange rates. Just because USD -> INR = 63 and INR -> GBP = 0.01
does not mean USD -> GBP = 0.63. It is actually 0.64.

File: OperatorOverloading/01 Operator Overloading/OperatorOverloading.Host/CurrencyConversion.cs [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+                }
+                Console.WriteLine("Enter Currency");
+                sourceCurrency = Console.ReadLine();
+
+                Money money1 = new Money(money,sourceCurrency);
+                
+                Console.WriteLine("\nEnter target Currency : ");
+                string target = Console.ReadLine();
+                Console.WriteLine("\nExchanged Amount : ");
+                var money2 = new Money(target);
```
Comment (ytokas) [inline OperatorOverloading/01 Operator Overloading/OperatorOverloading.Host/CurrencyConversion.cs:32]: why do you pass currency to money object?

File: OperatorOverloading/01 Operator Overloading/OperatorOverloading.Model/Money.cs [hunk 5]
```diff
@@ -3,25 +3,34 @@
...
+            {
+                _amount = value;
+            }
         }
         public string Currency
         {
             set
             {
-                if (string.IsNullOrWhiteSpace(value) || value.Length != 3)
+                if (string.IsNullOrEmpty(value) || value.Length != 3)
```
Comment (ytokas) [inline OperatorOverloading/01 Operator Overloading/OperatorOverloading.Model/Money.cs:31]: Why did you change it to IsNullOrEmpty ?

File: OperatorOverloading/01 Operator Overloading/OperatorOverloading.Model/Money.cs [hunk 6]
```diff
@@ -3,25 +3,34 @@
...
                 {
-                    throw new Exception(Messages.CurrencyEmpty);
+                    throw new Exception();
```
Comment (ytokas) [inline OperatorOverloading/01 Operator Overloading/OperatorOverloading.Model/Money.cs:33]: While throwing exception make sure you give proper message

File: OperatorOverloading/01 Operator Overloading/OperatorOverloading.Model/Money.cs [hunk 7]
```diff
@@ -38,39 +47,52 @@ public Money(double money, string tempCurrency)
         public Money()
         {
         }
+
+        public Money(string source)
+        {
+            this.Currency = source;
+        }
+
+        FileExchangeRateProvider fileExchangeRateProvider = new FileExchangeRateProvider();
```
Comment (ytokas) [inline OperatorOverloading/01 Operator Overloading/OperatorOverloading.Model/Money.cs:56]: Why is FileExchageRateProvider a property of Money class?

File: OperatorOverloading/01 Operator Overloading/OperatorOverloading.Model/Money.cs [hunk 8]
```diff
@@ -38,39 +47,52 @@ public Money(double money, string tempCurrency)
...
+
+        public  double Convert(double amount, string sourceCurrency, string targetCurrency)
+        {
+           
+            double exchangeRate = fileExchangeRateProvider.GetExchangeRate(sourceCurrency, targetCurrency);
+            return exchangeRate * amount;
+        }
+
         //operator + is overloaded
         public static Money operator +(Money Money1, Money Money2)
```
Comment (ytokas) [inline OperatorOverloading/01 Operator Overloading/OperatorOverloading.Model/Money.cs:67]: Again naming convention is wrong. Your assignment is rejected.
